### ISSUE TITLE
eval on COCO or ADE20k

### ISSUE THREAD

**Issue by shawnricecake** [YuYue525/MobileSAM-pytorch#16]
Hi,

Great work!

Is there any way to evaluate the results on COCO or ADE20k datasets?

Thanks

**Comment by shawnricecake** [comment]
by the way,

I notice that the evaluation you provide is still using the SA-1K dataset (exactly, some of them).

Is that the same way described in the mobilesam paper?

could you explain this?

Thanks

**Comment by shawnricecake** [comment]
I implement the evaluation on COCO dataset with nanosam (nvidia)

I would like to ask that if we can improve the performance by training with more data?
did you try it?

**Comment by YuYue525** [comment]
Great work! 
We can improve the performance by training with more data. > 8% SA-1B dataset, 8 epochs (around 225,000 iteration), 8 NVIDIA V100 16G GPUs with batch size = 32, the training time is around 28.5 hours ![1704268086645](https://private-user-images.githubusercontent.com/58425611/293858066-09c62c24-deab-4c8f-9422-b18dda2a33eb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyODU2MjgsIm5iZiI6MTcwNTI4NTMyOCwicGF0aCI6Ii81ODQyNTYxMS8yOTM4NTgwNjYtMDljNjJjMjQtZGVhYi00YzhmLTk0MjItYjE4ZGRhMmEzM2ViLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE1VDAyMjIwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ5NDI0OWU3OWFmZTg4NGI0MTU5ZTAzMjFjY2NlMTk3MmQ2NzY0ZmViMWU2NTY0NmQ2MGY5ZmYzMmM1YWZkZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wB9EXCGPb2m318cpLwSdhNYkqPMYCCBpQEMF9mB5ky0)

**Comment by shawnricecake** [comment]
How about the task performance?

**Comment by SunIsland-L** [comment]
I understand that someone has used the same settings of MobileSAM to achieve the accuracy in the paper, but the performance of your reference code with the same settings is poor. What is the problem? Thanks for the answer.

**Comment by shawnricecake** [comment]
What do you mean by achieving the accuracy in the paper? What’s that about?
On Coco?
And I just load the weights provided in this repo on coco to get about 73
which is poor, I don’t whether it can be improved when there are more
training data, if you can provide the weights trained with more data?

**Comment by SunIsland-L** [comment]
The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.

**Comment by shawnricecake** [comment]
Ok so the test acc on your side is with the SA dataset? Based on the
evaluation method provided in this repo?

**Comment by YuYue525** [comment]
This repo mainly provided the training framework of MobileSAM. We didn’t try too much on the choice of hyperparameters during training. So, there is still room for improvement. If you want to try to get better model performance, try changing the hyperparameters during training or introduce other tricks. 

In my experiment, I tried the same amount of training data as in the paper (1% SA-1B), but the model effect obtained was not ideal. Therefore, I provide a model weight obtained by training on more training sets (8% SA-1B) for your reference.

You can choose any metric to test on any data set as you like.

Thank you!

**Comment by shawnricecake** [comment]
Oh ok, so the weights you provided is trained with 8% dataset? That’s all
right. Actually I find that in the code , it seems that you you used more
than SA000001 - SA000009.
But never mind, Korean paper, dddd

**Comment by shawnricecake** [comment]
> The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.

Hi, it looks like you have trained the model on only 1% data? and got 69% iou?

**Comment by SunIsland-L** [comment]
> > The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.
> 
> Hi, it looks like you have trained the model on only 1% data? and got 69% iou?

Yes. Have you tried training? What was the result?

**Comment by jwfanDL** [comment]
> > The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.
> 
> Hi, it looks like you have trained the model on only 1% data? and got 69% iou?

And I'll be appreciate if you can provide your scripts that can evaluate on COCO dataset

**Comment by shawnricecake** [comment]
> > > The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.
> > 
> > 
> > Hi, it looks like you have trained the model on only 1% data? and got 69% iou?
> 
> And I'll be appreciate if you can provide your scripts that can evaluate on COCO dataset

You can simply get the code from [nanosam](https://github.com/NVIDIA-AI-IOT/nanosam) built by nvidia :)

**Comment by jwfanDL** [comment]
> > > > The officially provided .pth can reach 73% when tested on the SA-1K dataset. If the current training code is trained on the 1% SA-1B data set, the test accuracy is 69%.
> > > 
> > > 
> > > Hi, it looks like you have trained the model on only 1% data? and got 69% iou?
> > 
> > 
> > And I'll be appreciate if you can provide your scripts that can evaluate on COCO dataset
> 
> You can simply get the code from [nanosam](https://github.com/NVIDIA-AI-IOT/nanosam) built by nvidia :)

Got it with many thanks!
